hazelcast/src/main/java/com/hazelcast/map/impl/BasicMapContextQuerySupport.java

```diff
@@ -42,13 +44,17 @@
 
     private final MapServiceContext mapServiceContext;
     private final NodeEngine nodeEngine;
+    private final long maxLocalPartitionsLimitForPreCheck;
 
     public BasicMapContextQuerySupport(MapServiceContext mapServiceContext) {
         this.mapServiceContext = mapServiceContext;
         this.nodeEngine = mapServiceContext.getNodeEngine();
+        this.maxLocalPartitionsLimitForPreCheck = Math.abs(
+                nodeEngine.getGroupProperties().QUERY_MAX_LOCAL_PARTITION_LIMIT_FOR_PRE_CHECK.getInteger());
```

Comment:
Do we need to handle special case for `Integer.MIN_VALUE` when using `Math.abs` ?